bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -136,15 +140,45 @@ private Optional<Issue> getCsrIssue(Issue issue) {
         if (issueProject == null) {
             return Optional.empty();
         }
-        var jbsIssue = issueProject.issue(issue.shortId());
-        if (jbsIssue.isEmpty()) {
+        var jbsIssueOpt = issueProject.issue(issue.shortId());
+        if (jbsIssueOpt.isEmpty()) {
             return Optional.empty();
         }
-        var csr = csrLink(jbsIssue.get()).flatMap(Link::issue);
-        if (csr.isPresent()) {
-            return Issue.fromStringRelaxed(csr.get().id() + ": " + csr.get().title());
+
+        var jbsIssue = jbsIssueOpt.get();
+        var csrOpt = csrLink(jbsIssue).flatMap(Link::issue);
+        if (csrOpt.isEmpty()) {
+            return Optional.empty();
         }
-        return Optional.empty();
+
+        var csr = csrOpt.get();
+        if (pr.labelNames().contains("backport")) {
+            return findBackportCsr(jbsIssue, csr).flatMap(perIssue -> Issue.fromStringRelaxed(perIssue.id() + ": " + perIssue.title()));
+        }
+        return Issue.fromStringRelaxed(csr.id() + ": " + csr.title());
+    }
+
+    /**
+     * Find the right CSR according to the primary issue and the primary CSR
+     */
+    private Optional<org.openjdk.skara.issuetracker.Issue> findBackportCsr(org.openjdk.skara.issuetracker.Issue primary, org.openjdk.skara.issuetracker.Issue csr) {
+        var confFile = pr.repository().fileContents(".jcheck/conf", pr.targetRef());
+        var configuration = JCheckConfiguration.parse(confFile.lines().toList());
+        var version = configuration.general().version().orElse(null);
+        if (version == null || "".equals(version)) {
+            return Optional.empty();
+        }
+        var jdkVersion = JdkVersion.parse(version);
+        if (jdkVersion.isEmpty()) {
+            return Optional.empty();
+        }
```

Comment:
This block of code is repeated 3 times now. I know we don't have a good place to share it between different bots, but we should be able to put it in a single location for CSRCommand and CheckRun at least.